### ISSUE TITLE
🐛 <jemalloc>: Unsupported system page size

### ISSUE THREAD

**Issue by vhscom** [biomejs/biome#1692]
### Environment information

```block
bunx @biomejs/biome range
<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
memory allocation of 5 bytes failed
```

### What happened?

1. [Clone reproduction](https://codeberg.org/vhs/metaform/src/branch/biomejs-jemalloc-error) which adds Biome to a fresh Bun app.
2. Run `bun install && bun lint` to run the Node binary installed as a dev dependency.
3. Observe `<jemalloc>` error on `aarch64` systems.

### Expected result

Formatted code in `src` directory in accordance with specified Biome configuration. Instead, I see the following error:

![Screenshot from 2024-01-27 17-00-15](https://github.com/biomejs/biome/assets/97140109/e690490b-1013-469d-8473-3140bd0f8062)

Biome version: 1.5.3
Bun version: 1.0.21
Node version: 20.10.0
Linux kernel: 6.6.3-411.asahi.fc39.aarch64+16k

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
You should report the bug in the bun repository, as it aims to have Node.js compatibility. npx works fine.

**Comment by vhscom** [comment]
I don't use `npx` in my workflow. I'll revisit Biome.js compatibility with Bun later. The repro I will leave for now.

**Comment by ematipico** [comment]
It was an example. If npx works (or pnpm dlx, yarn dlx), which is based on Node.js, so should bunx.

Jarred explicitly said that incompatibility bugs with Node.js should be reported in the bun repository.

**Comment by jnlmmln** [comment]
Hi,

I have the same issue using `npx`:

```
❯ npx @biomejs/biome format .

<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
memory allocation of 5 bytes failed
```

Manually editing `node_modules/@biomejs/biome/bin/biome` and letting `isMusl()` always return true, so the `@biomejs/cli-linux-arm64-musl/biome` gets used is working.

node: v18.19.0
biome: 1.5.3
kernel: 6.6.3-413.asahi.fc39.aarch64+16k

**Comment by ematipico** [comment]
Probably your OS doesn't fit the heuristic of the function. Not sure what we should update to match your OS

**Comment by jnlmmln** [comment]
Thanks for the quick reply.
Reading through some issues the problem should be with jemallocator. Those should be related:

https://github.com/gnzlbg/jemallocator/issues/68
https://github.com/gnzlbg/jemallocator/issues/170

---

And setting the page size for aarch64 builds to 16 should fix it.
```
JEMALLOC_SYS_WITH_LG_PAGE=16
```

https://github.com/astral-sh/ruff/blob/af6ea2f5e4b1702d5062831624492d94f57a0f81/.github/workflows/release.yaml#L247

---

I would be happy to support on this.

**Comment by ematipico** [comment]
Would you like to send a PR?

**Comment by jnlmmln** [comment]
Sure: https://github.com/biomejs/biome/pull/1975
